### PR TITLE
ARROW-4175: [GLib] Add support for decimal compare operators

### PR DESCRIPTION
**Pull request by shiro615**

### CODE REVIEW

File: c_glib/arrow-glib/decimal128.cpp [hunk 1]
```diff
@@ -154,6 +154,98 @@ garrow_decimal128_equal(GArrowDecimal128 *decimal,
...
+garrow_decimal128_less_than_or_equal(GArrowDecimal128 *decimal,
+                                     GArrowDecimal128 *other_decimal)
+{
+  const auto arrow_decimal = garrow_decimal128_get_raw(decimal);
+  const auto arrow_other_decimal = garrow_decimal128_get_raw(other_decimal);
+  return *arrow_decimal <= *arrow_other_decimal;
+}
+
+/**
+ * garrow_decimal128_more_than:
```
Comment (kou) [inline c_glib/arrow-glib/decimal128.cpp]: Could you use `greater_than`?

File: c_glib/test/test-decimal128.rb [hunk 2]
```diff
@@ -106,4 +106,81 @@ def test_divide_zero
       decimal1.divide(decimal2)
     end
   end
+
+  def test_not_equal
```
Comment (kou) [inline c_glib/test/test-decimal128.rb]: Could you also add a test for `equal`?
(I didn't add it. Sorry.)

**Review by kou — CHANGES_REQUESTED** [review]
Thanks!
Could you check my comments?

File: c_glib/arrow-glib/decimal128.cpp (hunk 1 above)
Comment (shiro615) [reply c_glib/arrow-glib/decimal128.cpp]: Thank you for the pointing out. I'll fix it.

File: c_glib/test/test-decimal128.rb (hunk 2 above)
Comment (shiro615) [reply c_glib/test/test-decimal128.rb]: Yes. I'll add it.

**Review by kou — APPROVED** [review]
+1
I'll fix some documents and merge this.

**Comment by shiro615** [comment]
Thanks!

**Pull request closed without merging**